.github/workflows/test.yml

```diff
@@ -0,0 +1,26 @@
+---
+name: Test
+
+on:
+  push:
+    branches:
+    - "*"
+    tags-ignore:
+    - "*"
+  pull_request:
+    types: [opened, reopened]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2-beta
+      with:
+        go-version: '^1.14.1'
+    - uses: actions/cache@v1
+      with:
+        path: /home/runner/go/pkg/mod
+        key: go-mod
+    - name: Run test suite
+      run: make tests
```

Comment:
Why are you adding this workflow? Unit tests are already integrated into the build workflow: https://github.com/appuio/seiso/blob/master/.goreleaser.yml#L6

Reply:
TBH, I didn't realize this was there. And this is okay in some way. Tests are always executed implicitly before a release. That's somewhat also the philosophy of the go-releaser tool, IIUC, to do everything.

OTOH, it is less transparent, because there is no separate test job in the pipeline. This is not optimal.

Would you mind keeping the overhead (it's not a big one) for the sake of increased transparency? Otherwise, I'm okay with removing the test workflow again. No hard feelings.

Reply:
Well, the Build workflow is basically testing the whole release pipeline as far it can go.

I think adding the additional workflow would make it worse. Instead of 1 red failing check in a PR displayed on Github, a user would have to look up the CI job logs from at least 2 checks, only to find out that they were caused by the same failing unit test.

May I suggest a counter proposal where we rename the Job name in the Build workflow from `goreleaser` to `test`? That would make it a little better I think:
![image](https://user-images.githubusercontent.com/12159026/80347418-92440c00-886c-11ea-958b-e3b7ee5b1cfc.png)

Reply:
I went through the [goreleaser documentation](https://goreleaser.com/) and the tutorials linked to from therein. I found no evidence of the `hooks` section being used for triggering linting and tests. Even to the contrary, the documentation shows CI pipeline implementations in great lengths. The authors' intention seems clear, they want to separate the responsibilities of goreleaser (i.e. "releasing") and the CI services (to perform actions that support, prepare the act of releasing).

While it's _possible_ to Including linting, security scans and test execution in the release configuration we really shouldn't do that, unless we plan to release manually (which is, we likely agree, what we want to avoid since this lacks an audit trail). The good news is that, when changing this, we're both following the habits (suggestions?) of the authors as well as pipeline best practices. And that makes our project more maintainable in the long run.

Reply:
Do you have links to relevant documentation (not a generic one) or issues for these statements?